src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,683 @@
+/** Histogram Visualization. */
+loadScript("https://d3js.org/d3.v4.min.js");
+loadStyle("https://fontlibrary.org/face/dejavu-sans-mono");
+
+let shortcuts = {
+  zoomIn: (e) => (e.ctrlKey || e.metaKey) && e.key === "z",
+  showAll: (e) => (e.ctrlKey || e.metaKey) && e.key === "a",
+};
+
+const LABEL_STYLE = "font-family: DejaVuSansMonoBook; font-size: 10px;";
+const X_AXIS_LABEL_WIDTH = 30;
+const ANIMATION_DURATION = 1000;
+const LINEAR_SCALE = "linear";
+const LIGHT_PLOT_COLOR = "#00E890";
+const DARK_PLOT_COLOR = "#E0A63B";
+const DEFAULT_NUMBER_OF_BINS = 10;
+const BUTTON_HEIGHT = 25;
+
+// Indicates whether script dependencies have been added to the DOM.
+let scriptsInitialised = false;
+
+/**
+ * A d3.js histogram visualization.
+ *
+ *
+ * Data format (json):
+ {
+   "axis" : {
+      "x" : { "label" : "x-axis label", "scale" : "linear" },
+      "y" : { "label" : "y-axis label", "scale" : "logarithmic" },
+   },
+   "focus" { "x" : 1.7, "y" : 2.1, "zoom" : 3.0 },
+   "color" :  "rgb(1.0,0.0,0.0)" },
+   "bins" : 10,
+   "data" : [
+     "values" : [0.1, 0.2, 0.1, 0.15, 0.7],
+   ]
+ }
+ */
+class Histogram extends Visualization {
+  static inputType = "Any";
+  static label = "Histogram";
+
+  onDataReceived(data) {
+    const parsedData = JSON.parse(data);
+    const isUpdate = parsedData.update === "diff";
+    this.updateState(parsedData, isUpdate);
+
+    if (!this.isInitialised()) {
+      this.initCanvas();
+      this.initLabels();
+      this.initHistogram();
+    }
+    this.updateLabels();
+    this.updateHistogram();
+  }
+
+  /**
+   * Indicates whether this visualisation has been initialised.
+   */
+  isInitialised() {
+    ok(this.svg);
+  }
+
+  /**
+   * Update the internal data and plot settings with the ones from the new incoming data.
+   * If no new settings/data have been provided the old ones will be kept.
+   */
+  updateState(data, isUpdate) {
+    if (isUpdate) {
+      this._axisSpec = ok(data.axis) ? data.axis : this._axisSpec;
+      this._focus = ok(data.focus) ? data.focus : this._focus;
+      this._dataValues = ok(data.data.values) ? data.data.values : this.data;
+      this._bins = ok(data.bins) ? data.bins : this._bins;
+    } else {
+      this._axisSpec = data.axis;
+      this._focus = data.focus;
+      this._dataValues = data.data.values;
+      this._bins = data.bins;
```

Comment:
are you using some kind of formatter here? I was asking to use prettier during osme previosu comments with disabled semicolon option